Minor UI tweaks for the watching interface!

DigitalDerrickcs opened the pull request.

ghost commented:
# Optic PR Check
_Last updated @ 2:56pm UTC on March 3, 2021 UTC_


Hey! Looks like you do not have Optic tracking this API’s specification. Set up Optic, then check-in a spec to use the bot. [Read the Optic Docs](https://www.useoptic.com/docs/)


#### Powered by [Optic](https://www.useoptic.com). [Not seeing changes?](https://www.useoptic.com/docs/documenting-your-api/)

acbart committed:
Give the containers names

acbart committed:
Debug info needed for redis

acbart committed:
Connect nginx through 8888 port instead

acbart committed:
Longer connection for uwsgi

acbart committed:
Really long connection times

acbart committed:
Ensure proper protocol

acbart committed:
Debugging support

acbart committed:
Link application to wsgi correctly

acbart committed:
Dump uwsgi.ini file

acbart committed:
Check permissions I guess

acbart committed:
More logs

acbart committed:
Less logs

acbart committed:
Kill that template

acbart committed:
A little more debugging

acbart committed:
Force python3

acbart committed:
Throwing stuff at the wall

acbart committed:
Why is this wrong?

acbart committed:
Need to specify the instance folder more explicitly

acbart committed:
Change to psycopg see if that works

acbart committed:
More python debug output

acbart committed:
Really try to force the python version

acbart committed:
Can we just upgrade the python version?

acbart committed:
Oops forgot the site packages

acbart committed:
Forget the site packages for a second

acbart committed:
Force the home folder for python

acbart committed:
Use a more real python

acbart committed:
Try this instead

acbart committed:
Let's try gunicorn instead

acbart committed:
Wrong port!

acbart committed:
Logs needed

acbart committed:
Most fields CAN be optional

acbart committed:
Hopefully copy with correct permissions

acbart committed:
I need less

acbart committed:
Update nginx for gunicorn

acbart committed:
Update nginx for gunicorn

acbart committed:
Escape $ for docker

acbart committed:
Escape $ for docker

acbart committed:
Debug output

acbart committed:
Debug output

acbart committed:
Change the nginx file back

acbart committed:
Escape $

acbart committed:
Update SQL notes

acbart committed:
Allow access to backups from db

acbart committed:
Some cleanup of debug stuff

acbart committed:
Prevent type casting in courses

acbart committed:
Prevent type casting in courses

acbart committed:
Prevent type casting in courses

acbart committed:
Prevent type casting in courses

acbart committed:
New skulpt, pedal, designer

acbart committed:
Remove debug information

acbart committed:
Create shortcut URL for assignment list page

acbart committed:
Fix maze toolbar

acbart committed:
Stability improvements to Kettle's execution environment

acbart committed:
Allow summary feedback for exams

acbart committed:
Fix accessing history logs

acbart committed:
Forking menu!

acbart committed:
Forking menu now allows you to choose the assignment group

acbart committed:
Update skulpt and blockpy

acbart committed:
Make late policy disablable

acbart committed:
Some updates to blockpy interface and skulpt pil

acbart committed:
Fix quiz resubmit button

acbart committed:
Prevent empty quizzes from scoring perfects

acbart committed:
Finish late policy code

acbart committed:
First part of code search interface

acbart committed:
Allow reviews to be given for non-manually reviewed assignments

acbart committed:
Push new drafter stuff!

acbart committed:
Oops

acbart committed:
Updated skulpt

acbart committed:
Fix function name for getpixel in skulpt pillow

acbart committed:
Positive feedback used as debugging tools

acbart committed:
Try to force HTTPS external URLs

acbart committed:
Force a specific URL to be https

acbart committed:
Force all external URLs to use https

acbart committed:
Final fixes for skulpt's drafter+matplotlib+pil

acbart committed:
Fix subtle blockpy bug

acbart committed:
Fix syntax errors in ast module

acbart committed:
Fix pedal's skulpt

acbart committed:
Pedal update

acbart committed:
Bulk transfer menu completed!

acbart committed:
Only show transferable old submissions in bulk transfer menu

acbart committed:
Limit code width in bulk transfer menu

acbart committed:
Limited R support

acbart committed:
Pedal updates

acbart committed:
Update pedal, drafter, and skulpt

acbart committed:
Update pedal, drafter, and skulpt

acbart committed:
Update pedal, drafter, and skulpt

acbart committed:
Fix for drafter bugs

acbart committed:
Updated skulpt too

acbart committed:
Minor bug fixes for drafter and pedal

acbart committed:
Update drafter and skulpt

acbart committed:
Drafter and blockpy improvements, such as link sharing

acbart committed:
Allow exporting explicit individual users

acbart committed:
Fix instructor check for editing

acbart committed:
Make json settings editable by the instructor

acbart committed:
Update blockly! Newest version, with joinedstr and other goodies.

acbart committed:
Include the english language

acbart committed:
Copied over new language files for blockly (unused so far)

acbart committed:
Updated the frontend to use newer node

acbart committed:
Minor fix for providing courseId to list-files in frontend

acbart committed:
Make log table use an estimated count in postgres

acbart committed:
Show type of assignment in manage assignments view

acbart committed:
Just turn off pagination on the log table

acbart committed:
Just turn off pagination on the log and submissions table

acbart committed:
QR Code Link

acbart committed:
Update Drafter

acbart committed:
Fixes for quizzes, new explains mode

acbart committed:
Don't need browserfs

acbart committed:
Unnecessary Import

acbart committed:
Hopefully setup alembic

acbart committed:
Starting to update the database to prism version

acbart committed:
Banned users fixed

acbart committed:
Merge branch 'master' into sum25

acbart committed:
First stab at backporting updates

acbart committed:
Temporarily disabled banning, secure fileadmin, progsnap updates

acbart committed:
Track IP addresses

acbart committed:
Handle escaped square brackets better, and also missing questions in analysis

acbart committed:
Merge branch 'master' into sum25

# Conflicts:
#	common/text.py
#	models/user.py

acbart committed:
Update models for prism updates

acbart committed:
Handle database changes

acbart committed:
Fix hiding for other report types

acbart committed:
Merge branch 'master' into sum25

acbart committed:
Add in feedback column, and user counts tables

acbart committed:
New tables in progress

acbart committed:
New timer feature

acbart committed:
Delete old js file

acbart committed:
Document admin index html file

acbart committed:
Remove youtube integration

acbart committed:
Officially remove java support

acbart committed:
Separate out CSS file from layout.html into frontend.css

acbart committed:
Remove doppio utils

acbart committed:
Extract out some frontend logic into frontend npm, and then also start organizing frontend/components better

acbart committed:
Finish reorganizing existing frontend components

acbart committed:
Small amount of code cleanup in editor.html

acbart committed:
Separate out the reader's html template from its logic

acbart committed:
Properly hide textbook editing in course view for non-instructors

acbart committed:
TAs can't see hidden assignments' submissions

acbart committed:
Notes for clearing/transferring the DB

acbart committed:
Don't show the quiz in the output

acbart committed:
Update skulpt, blockpy, pedal, and drafter

acbart committed:
Use strenum instead of enum.StrEnum

acbart committed:
Accidentally named the logs folder something git ignored!

acbart committed:
Fix logging for log_event

acbart committed:
Fix user_counts a little

acbart committed:
Hide unnecessary links in admin

acbart committed:
Make sure that TAs cannot access Hidden (exam) assignments

acbart committed:
Some unnecessary logging

acbart committed:
Update vendored deps

acbart committed:
Fix make_log_entry call

acbart committed:
Fix save_student_file

acbart committed:
Allow empty settings.

acbart committed:
Oops forgot to build

Copilot committed:
Initial plan

Copilot committed:
Implement test infrastructure and User/Course model tests

Co-authored-by: acbart <897227+acbart@users.noreply.github.com>

Copilot committed:
Complete comprehensive test suite with model, controller, and workflow tests

Co-authored-by: acbart <897227+acbart@users.noreply.github.com>

Copilot committed:
Final comprehensive test suite with documentation and fixes

Co-authored-by: acbart <897227+acbart@users.noreply.github.com>

acbart committed:
Handle rest of stuff for locking exams for time

acbart committed:
Fix kettle iframe

acbart committed:
Light type info on post_grade

acbart committed:
Change default history display

acbart committed:
History loading should enforce types

acbart committed:
Fix history loading for exam, also new system for displaying exam tags

acbart committed:
Get tag data from quizzes

acbart committed:
Fix pedal types

acbart committed:
Fixed logic in quiz showing

acbart committed:
Allow instructors to see the quiz feedback

acbart committed:
Fix LTI launches for feedback tags in quizzes

acbart committed:
More logging for errors with the timer

acbart committed:
Handle isInstructor check correctly

acbart committed:
Update make red flag report

acbart committed:
Better error reporting for the red flag report

acbart committed:
Improve error handling in red flag report generation

acbart committed:
Launch read-only editors with display seed set to submission ID

acbart committed:
Fix history rendering for new events, timestamps, and in alone mode

acbart committed:
Speed up loading all the submitted assignments for submissions_filter

acbart committed:
Don't try and make distinct fancy

acbart committed:
Make smartHandleDate handle correct timezones correctly

acbart committed:
Provide link to time management

acbart committed:
Fix emoji counting, improve RFR data

acbart committed:
Smarter initial value for handling events in duration until success

acbart committed:
Trying a faster diffing algorithm

acbart committed:
Don't include instructors

acbart committed:
TAs can edit their own non-hidden submission

acbart committed:
Update pedal, skulpt, and drafter

acbart committed:
Update drafter to 1.8.6

acbart committed:
Provide for Administrator role (distinct from Admin)

acbart committed:
Fix the grade queuing in tasks

acbart committed:
Hot fix to prevent database crashes

acbart committed:
Force strings to make log

acbart committed:
Disable autocomplete for quizzes

acbart committed:
Fixing tests to be consistently pytest

acbart committed:
Basic user testing

acbart committed:
Sample endpoint test

acbart committed:
Merge branch 'server-tests'

acbart committed:
Restructure main application as proper application factory for testing

acbart committed:
First tests for actually loading some sample data (courses and users)

acbart committed:
Allow direct login for tests

Copilot committed:
Initial plan

Copilot committed:
Add comprehensive course endpoint tests (52 new tests covering security and authorization)

Co-authored-by: acbart <897227+acbart@users.noreply.github.com>

acbart committed:
Fix logic error in courses.py and the grader dashboard

acbart committed:
Fix up some of the tests

acbart committed:
Setup testing data files for assignment, group, membership, and submission

Copilot committed:
Initial plan

Copilot committed:
Add test data and comprehensive test files for assignments and groups

Co-authored-by: acbart <897227+acbart@users.noreply.github.com>

Copilot committed:
Fix CSV data formats and update tests for actual API behavior

Co-authored-by: acbart <897227+acbart@users.noreply.github.com>

Copilot committed:
Add comprehensive test documentation and summary

Co-authored-by: acbart <897227+acbart@users.noreply.github.com>

acbart committed:
New tests, and fixes for various permission errors related to assignments and groups

acbart committed:
Updating skulpt and blockpy

acbart committed:
Remove old doppio and blockly-games source from repo

acbart committed:
Don't link in doppio to assets

acbart committed:
Updating event handling stuff for blockpy client, skulpt

acbart committed:
Gitkeep some of the subfolders for convenient setups

acbart committed:
Gitkeep some more of the subfolders for convenient setups

acbart committed:
Fix missing submission status enum

acbart committed:
Clarify self-signing certificates process for Windows

Added note about self-signing certificates on Windows.

acbart committed:
Config for vscode compatibility

acbart committed:
Formatting ts file

acbart committed:
Submission counts, start of role counts, tracking user logins

acbart committed:
Fix datetime handling for flask admin

acbart committed:
Time limits are now handled by group instead of assignment

acbart committed:
Make course pinning a local only thing

acbart committed:
Track event counters

acbart committed:
Remove some dead code

acbart committed:
Fixes for skulpt/blockpy

acbart committed:
Event tracking updates

acbart committed:
Merge branch 'master' of https://github.com/blockpy-edu/blockpy-server

acbart committed:
More robust eror handling in quizzes

acbart committed:
New event tracking system with more sophistication

acbart committed:
Minor typo tracking syntax and runtime errors

acbart committed:
Fix minor issue with type of parameter

acbart committed:
Correctly register window and timer watchers for all parts of the course

acbart committed:
Simple route for getting rfr data

acbart committed:
Submission metadata directly in rfr

acbart committed:
Fix bug in posting 0 grades

acbart committed:
Fix major bug in skulpt's int/float comparison operations

acbart committed:
Minor UI tweaks

acbart committed:
Dump metrics on individual submissions

acbart committed:
Fix metrics header order

acbart committed:
Fix submission count query

acbart committed:
Add in adoption explanation to readme

acbart committed:
Reformulate query for fake dashboard

acbart committed:
Turn off synchronized commits for counts

acbart committed:
Proxy fix, timer control

acbart committed:
Allow missing submissions in exam time manager

acbart committed:
Filter dashboard by users

acbart committed:
Allow single submission view for shared URLs

Copilot committed:
Add comprehensive Python tests for all quiz question types (70 tests)

Agent-Logs-Url: https://github.com/blockpy-edu/blockpy-server/sessions/ac8edbd5-522e-4d04-85c1-bafaab5c3c1b

Co-authored-by: acbart <897227+acbart@users.noreply.github.com>

Copilot committed:
Fix KO bugs, add quiz editor state/UI, wire into quizzer component

Agent-Logs-Url: https://github.com/blockpy-edu/blockpy-server/sessions/ac8edbd5-522e-4d04-85c1-bafaab5c3c1b

Co-authored-by: acbart <897227+acbart@users.noreply.github.com>

acbart committed:
Frocing quiz editor reading id

acbart committed:
CLI version of fake dashboard

acbart committed:
Less memory intensive pivot in fake-dashboard

acbart committed:
Faster added_chars in rfr v1

acbart committed:
Another faster path for comparing strings in rfr

acbart committed:
Forget the difflib aspect entirely

acbart committed:
Fix incorrect reference to STAFF_ROLES

acbart committed:
Handle progsnap unicode issue

acbart committed:
Fix csv library blank strings. Goodness knows what else.

acbart committed:
Studio v1

acbart committed:
Survey stuff setup

acbart committed:
Rebuilt frontend

acbart committed:
Merge branch 'master' into copilot/add-quiz-editor-feature

# Conflicts:
#	static/libs/blockpy_server/frontend.js
#	static/libs/blockpy_server/frontend.js.map

acbart committed:
Quizzer work

acbart committed:
LAPIS Feedback Alpha

acbart committed:
Don't include Lapis Feedback just yet

acbart committed:
Change wording for survey

acbart committed:
Update studio

acbart committed:
Just... the entire feedback system. From one prompt.

acbart committed:
Some security gaps fixed

acbart committed:
Security fixes

acbart committed:
Updating versions

acbart committed:
Bug in users

acbart committed:
Update copyright

acbart committed:
massive improvement to the assignment manager

acbart committed:
Further assignment interface improvements

acbart committed:
Further refinements to assignment manager

acbart committed:
Minor restructuring and index cleanup for models

acbart committed:
Various optimizations for queries

acbart committed:
Faster fake dashboard

acbart committed:
In progress commit for the new view submissions menu

acbart committed:
Search interface for view submission

acbart committed:
Fix for the time/date representation

acbart committed:
Improve score and correctness rendering

acbart committed:
Dashboard spec, for the future...

acbart committed:
Improve the search for view submissions

acbart committed:
New analytics page

acbart committed:
Fix survey grading

acbart committed:
Handle zero case

acbart committed:
Rebuilt site

acbart committed:
Small updates for studio, fix weird behavior with submissions filter

acbart committed:
Fix for launching urls from dashboard

acbart committed:
Update studio

acbart committed:
Update studio

acbart committed:
Allow exporting by url for group and course

acbart committed:
Fix import group check

acbart committed:
Update studio

acbart committed:
Block safari and macos chrome 150
